app/scripts/modules/core/src/instance/details/console/ConsoleOutputModal.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+
+import { IInstance } from '../../../domain';
+import { IModalComponentProps, ModalBody, ModalFooter, ModalHeader } from '../../../presentation';
+import { Spinner } from '../../../widgets';
+import { InstanceReader, IInstanceConsoleOutput, IInstanceMultiOutputLog } from '../../InstanceReader';
+
+import './ConsoleOutputModal.less';
+
+export interface IConsoleOutputModalProps extends IModalComponentProps {
+  instance: IInstance;
+  usesMultiOutput: boolean;
+}
+
+export const ConsoleOutputModal = ({ dismissModal, instance, usesMultiOutput }: IConsoleOutputModalProps) => {
+  const [loading, setLoading] = React.useState(true);
+  const [selectedLog, setSelectedLog] = React.useState(null);
+  const [consoleOutput, setConsoleOutput] = React.useState(null);
+  const [exception, setException] = React.useState(null);
+  const logContainerEnd = React.useRef(null);
+
+  const fetchLogs = () => {
+    setLoading(true);
+    InstanceReader.getConsoleOutput(instance.account, instance.region, instance.id, instance.provider).then(
+      (response: IInstanceConsoleOutput) => {
+        setConsoleOutput(response.output);
+        if (!selectedLog) {
+          setSelectedLog(response.output[0]);
+        }
+        setLoading(false);
+      },
+      (exception) => setException(exception),
+    );
+  };
+
+  const jumpToEnd = () => {
+    logContainerEnd.current.scrollIntoView({ behavior: 'smooth' });
+  };
+
+  React.useEffect(() => {
+    fetchLogs();
+  }, []);
```

Comment:
you could switch to `useData` to avoid manually managing some of that state

Reply:
great idea thanks!